projects/plugins/jetpack/extensions/shared/test/block-fixtures.js

```diff
@@ -0,0 +1,287 @@
+				} );
+				expect( errors.length ).toEqual( 0 );
+				if ( errors.length ) {
+					throw new Error( 'Problem(s) with fixture files:\n\n' + errors.join( '\n' ) );
+				}
+			} );
+		}
+	} );
+}
+/* eslint-disable jest/no-export */
+
+function registerBlocks( blocks ) {
+	// Need to add a valid category or block registration fails
+	setCategories( [
+		{
+			slug: 'test',
+			title: 'Test',
+		},
+	] );
+	blocks.forEach( block => {
+		registerBlockType( block.name, { ...block.settings, category: 'test' } );
+	} );
+}
+
+function normalizeParsedBlocks( blocks ) {
+	return blocks.map( ( block, index ) => {
+		// Clone and remove React-instance-specific stuff; also, attribute
+		// values that equal `undefined` will be removed. Validation issues
+		// add too much noise so they get removed as well.
+		block = JSON.parse( JSON.stringify( omit( block, 'validationIssues' ) ) );
+
+		// Change client IDs to a predictable value
+		block.clientId = '_clientId_' + index;
+
+		// Recurse to normalize inner blocks
+		block.innerBlocks = normalizeParsedBlocks( block.innerBlocks );
+
+		return block;
+	} );
+}
+
+function readFixtureFile( fixturesDir, filename ) {
+	try {
+		return fs.readFileSync( path.join( fixturesDir, filename ), 'utf8' );
+	} catch ( err ) {
+		return null;
+	}
+}
+
+function writeFixtureFile( fixturesDir, filename, content ) {
+	fs.writeFileSync( path.join( fixturesDir, filename ), content );
+}
+
+function setFixturesDir( fixturePath ) {
+	FIXTURES_DIR = path.join( fixturePath, 'fixtures' );
+}
+function blockNameToFixtureBasename( blockName ) {
+	return blockName.replace( /\//g, '__' );
+}
+
+function getAvailableBlockFixturesBasenames() {
+	// We expect 4 different types of files for each fixture:
+	//  - fixture.html            : original content
+	//  - fixture.parsed.json     : parser output
+	//  - fixture.json            : blocks structure
+	//  - fixture.serialized.html : re-serialized content
+	// Get the "base" name for each fixture first.
+	return uniq(
+		fs
+			.readdirSync( FIXTURES_DIR )
+			.filter( f => /(\.html|\.json)$/.test( f ) )
+			.map( f => f.replace( /\..+$/, '' ) )
+	);
+}
+
+function getBlockFixtureHTML( basename ) {
+	const filename = `${ basename }.html`;
+	return {
+		filename,
+		file: readFixtureFile( FIXTURES_DIR, filename ),
+	};
+}
+
+function getBlockFixtureJSON( basename ) {
+	const filename = `${ basename }.json`;
+	return {
+		filename,
+		file: readFixtureFile( FIXTURES_DIR, filename ),
+	};
+}
+
+function getBlockFixtureParsedJSON( basename ) {
+	const filename = `${ basename }.parsed.json`;
+	return {
+		filename,
+		file: readFixtureFile( FIXTURES_DIR, filename ),
+	};
+}
+
+function getBlockFixtureSerializedHTML( basename ) {
+	const filename = `${ basename }.serialized.html`;
+	return {
+		filename,
+		file: readFixtureFile( FIXTURES_DIR, filename ),
+	};
+}
+
+function writeBlockFixtureHTML( basename, fixture ) {
+	writeFixtureFile( FIXTURES_DIR, `${ basename }.html`, fixture );
+}
```

Comment:
I don't think this is being used anywhere. Maybe I missed something but if not, perhaps we should remove it?
```suggestion
```

Reply:
Removed